tests/integration/device_test.sh

```diff
@@ -4,87 +4,261 @@ include './tests/unit/utils.sh'
+    container="kw-${distro}"
+
+    # Get the output of the `kw device --local` command from the container
+    container_output=$(container_exec "$container" 'kw device --local')
+    # Construct a regex by joining the filter items with a | separator
+    filter_regex=$(printf '%s|' "${output_filter[@]}" | sed 's/|$//')
+    # The actual perl regex.
+    filter_regex="(${filter_regex}):"
+    # Get the output, after applying filter.
+    output=$(printf '%s' "${container_output}" | grep --invert-match --perl-regexp "$filter_regex")
+    # Store the result in the array
+    results["$distro"]="$output"
+  done
+}
+
+function test_kw_device_local_ram()
+{
+  local distro
+  local container
+  local mem_total_kb
   local expected_output
+  local output
 
-  # buffer with all information.
-  container="kw-${distro}"
-  buffer=$(container_exec "${container}" 'kw device --local')
+  for distro in "${!results[@]}"; do
+    container="kw-${distro}"
+    output="${results[$distro]}"
+    # Get the RAM value in kilobytes from the container
+    mem_total_kb=$(container_exec "$container" "awk '/MemTotal/ {print \$2}' /proc/meminfo")
+    # Convert the RAM value to IEC format
+    expected_output=$(container_exec "$container" "numfmt --from=si --to=iec ${mem_total_kb}K")
+    # Extract RAM value from the output
+    output=$(printf "%s" "$output" | grep "Total RAM" | awk '{print $3}')
+    # Assert that the expected RAM value matches the output RAM value
+    assert_equals_helper "'kw device' RAM check failed for ${distro}" "$LINENO" "$expected_output" "$output"
+  done
+}
 
-  # some fields must be ignored because they surely won't match.
-  filter=(
-    'Root filesystem' 'Size' 'Mounted on'                                            # storage fields
-    'Distribution' 'Distribution base' 'Distribution version' 'Desktop environments' # desktop fields
-  )
+function test_kw_device_local_storage()
+{
+  local distro
+  local container
+  local output_fs
+  local output_size
+  local output_mounted
+  local expected_fs
+  local expected_size
+  local expected_mounted
+
+  for distro in "${!results[@]}"; do
+    container="kw-${distro}"
+
+    # Retrieve expected root filesystem from container
+    expected_fs=$(container_exec "$container" "df -h / | tail -n 1 | awk '{print \$1}'")
+    # Extract actual root filesystem from output
+    output_fs=$(printf "%s" "${results[$distro]}" | awk '/Root filesystem:/ {print $3}')
+    # Assert that expected and actual root filesystem values match
+    assert_equals_helper "'kw device' Root filesystem mismatch for ${distro}" "$LINENO" "$expected_fs" "$output_fs"
+
+    # Retrieve expected root filesystem size from container
+    expected_size=$(container_exec "$container" "df -h / | tail -n 1 | awk '{print \$2}'")
+    # Extract actual root filesystem size from output
+    output_size=$(printf "%s" "${results[$distro]}" | awk '/Size:/ {print $2}')
+    # Assert that expected and actual root filesystem size values match
```

Comment:
I'm under the impression that all of these comments could be used to create a better variable name and, with that, get rid of the comments.